gateway-sp-comms/src/single_sp.rs

```diff
@@ -740,10 +750,67 @@ impl SingleSp {
     /// Instruct the SP to reset a component.
     ///
     /// Only valid after a successful call to `reset_component_prepare()`.
+    ///
+    /// If `disable_watchdog` is `true`, then any watchdogs associated with the
+    /// reset are disabled.  Otherwise, watchdogs are enabled opportunistically
+    /// (depending on component and MGS protocol version).
     pub async fn reset_component_trigger(
         &self,
         component: SpComponent,
+        disable_watchdog: bool,
     ) -> Result<()> {
+        // MGS protocol version in which SP watchdog messages were added
+        const MGS_WATCHDOG_VERSION: u32 = 12;
+
+        // If the SP has an update pending, then use the watchdog reset
+        let mut use_watchdog = component == SpComponent::SP_ITSELF
+            && !disable_watchdog
+            && matches!(
+                self.update_status(component).await?,
+                UpdateStatus::Complete(..)
```

Comment:
Is this check strictly required for correctness, or would it be okay to use the watchdog even for non-update resets? If it is required, we might have a similar problem as #141 in that this is inherently racy, and some other client could reset the SP in between this check and the subsequent requests this method makes. If it's not required, I'd be tempted to remove this check, because (a) I think the vast majority of resets are probably to apply an update status and (b) it removes something that looks like a TOCTOU race.

Reply:
See https://github.com/oxidecomputer/management-gateway-service/pull/216#discussion_r1566297077 ; this check is for ergonomics, but the SP also does the same check atomically as part of applying the `ResetComponentTriggerWithWatchdog` request.